cli/pcluster/config_sanity.py

```diff
@@ -25,424 +25,457 @@
-        for item in mt.get("MountTargets"):
-            # Check to see if there is an existing mt in the az of the stack
-            mt_subnet = item.get("SubnetId")
-            if availability_zone == ec2.describe_subnets(SubnetIds=[mt_subnet]).get("Subnets")[0].get(
-                "AvailabilityZone"
-            ):
-                mt_id = item.get("MountTargetId")
-        # If there is an existing mt in the az, need to check the inbound and outbound rules of the security groups
-        if mt_id:
-            nfs_access = False
-            in_access = False
-            out_access = False
-            # Get list of security group IDs of the mount target
-            sg_ids = efs.describe_mount_target_security_groups(MountTargetId=mt_id).get("SecurityGroups")
-            for sg in ec2.describe_security_groups(GroupIds=sg_ids).get("SecurityGroups"):
-                # Check all inbound rules
-                in_rules = sg.get("IpPermissions")
-                for rule in in_rules:
-                    if check_sg_rules_for_port(rule, 2049):
-                        in_access = True
-                        break
-                out_rules = sg.get("IpPermissionsEgress")
-                for rule in out_rules:
-                    if check_sg_rules_for_port(rule, 2049):
-                        out_access = True
-                        break
-                if in_access and out_access:
-                    nfs_access = True
-                    break
-            if not nfs_access:
-                _fail(
-                    "EFSFSId",
-                    "There is an existing Mount Target %s in the Availability Zone %s for EFS %s, "
-                    "and it does not have a security group with inbound and outbound rules that support NFS. "
-                    "Please modify the Mount Target's security group, or delete the Mount Target."
-                    % (mt_id, availability_zone, resource_value[0])
-                )
-
-            return True
-    except ClientError as e:
-        _fail("EFSFSId", e.response.get("Error").get("Message"))
+    @staticmethod
+    def __check_sg_rules_for_port(rule, port_to_check):
+        port = rule.get("FromPort")
+        ip_rules = rule.get("IpRanges")
+        group = rule.get("UserIdGroupPairs")
 
+        for ip_rule in ip_rules:
+            ip = ip_rule.get("CidrIp")
+            # An existing rule is valid for EFS if, it allows all traffic(0.0.0.0/0)
+            # from all ports or NFS(port 2049), and does not have a security group restriction
+            if (not port or port == port_to_check) and ip == "0.0.0.0/0" and not group:
+                return True
 
-def check_resource(region, aws_access_key_id, aws_secret_access_key, resource_type, resource_value):  # noqa: C901 FIXME
-    # Loop over all supported resource checks
-    # EC2 KeyPair
-    if resource_type == "EC2KeyPair":
+    def __check_efs_fs_id(self, ec2, efs, resource_value):  # noqa: C901 FIXME!!!
```

Comment:
Same here: not all paths return a value.

Reply:
Fixed by removing the return at all.